app/routers/event.py

```diff
@@ -551,6 +569,51 @@ def add_new_event(values: dict, db: Session) -> Optional[Event]:
         return None
 
 
+def extract_shared_list_from_data(
+    event_info: ImmutableMultiDict,
+    db: Session,
+) -> Optional[SharedList]:
+    """Extract shared list items from POST data.
+    Return:
+        SharedList: SharedList object stored in the database.
+    """
+
+    items = zip(
+        event_info.getlist("item-name"),
+        event_info.getlist("item-amount"),
+        event_info.getlist("item-participant"),
+    )
+    shared_list = {"title": event_info.get("shared-list-title"), "items": []}
```

Comment:
This is a bit of a complex structure, you can greatly simplify it and your code will look much better.

```python
title = event_info.get("shared-list-title")
items = []
...
return created_shared_list({title: items}, db)
```

Always prefer flat structures :)

Reply:
Actually, the reason I though it was best to write it this way is that I thought that accessing an unknown key (and it's value) is a bit of a brute-force. But I also agree, so I implemented your suggestion on my code.
Thanks!